tests/phpunit/DomainManagerTests.php

```diff
@@ -0,0 +1,121 @@
+<?php
+namespace EAMann\Dynamic_CDN;
+
+use Mockery\Mock;
+use WP_Mock as M;
+
+/**
+ * Class DomainManager_Tests
+ *
+ * @package EAMann\Dynamic_CDN
+ *
+ * @runTestsInSeparateProcesses
+ */
+class DomainManager_Tests extends TestCase {
+	protected $testFiles = [
+		'classes/DomainManager.php'
+	];
+
+	public function test_companion_constructor() {
+		$first = new DomainManager( 'http://test.com' );
+		$second = DomainManager( 'http://test2.com' );
+		$third = DomainManager( 'http://test2.com' );
+
+		$this->assertNotSame( $second, $first );
+		$this->assertInstanceOf( '\EAMann\Dynamic_CDN\DomainManager', $first );
+		$this->assertInstanceOf( '\EAMann\Dynamic_CDN\DomainManager', $second );
+		$this->assertSame( $second, $third );
+	}
+
+	public function test_adds_domains() {
+		$manager = DomainManager( 'http://test.com' );
+
+		$this->assertTrue( $manager->add( 'http://cdn1.com', 'uploads' ) );
+		$this->assertTrue( $manager->add( 'http://cdn2.com', 'uploads' ) );
+		$this->assertFalse( $manager->add( 'http://cdn1.com', 'uploads' ) );
+	}
+
+	public function test_cdn_domain_selection() {
+		$manager = DomainManager( 'http://test.com' );
+		$manager->add( 'http://cdn1.com', 'uploads' );
+		$manager->add( 'http://cdn2.com', 'uploads' );
+		$manager->add( 'http://cdn3.com', 'uploads' );
+
+		$this->assertEquals( $manager->cdn_domain( 'file1.jpg' ), $manager->cdn_domain( 'file1.jpg' ) );
+		$this->assertNotEquals( $manager->cdn_domain( 'file1.jpg' ), $manager->cdn_domain( 'file2.jpg' ) );
+	}
+
+	public function test_domain_replacement() {
+		$manager = DomainManager( 'http://test.com' );
+		$manager->add( 'http://cdn1.com', 'uploads' );
+
+		$original = 'http://test.com/image.png';
+		$expected = 'http://cdn1.com/image.png';
+
+		// Mocks
+		M::wpFunction( 'is_ssl', [ 'return' => false ] );
+		M::wpPassthruFunction( 'esc_url' );
+
+		// Verify
+		$this->assertEquals( $expected, $manager->new_url( $original ) );
+	}
+
+	public function test_domain_replacement_ssl() {
+		$manager = DomainManager( 'https://test.com' );
+		$manager->add( 'https://cdn1.com', 'uploads' );
+
+		$original = 'https://test.com/image.png';
+		$expected = 'https://cdn1.com/image.png';
+
+		// Mocks
+		M::wpFunction( 'is_ssl', [ 'return' => true ] );
+		M::wpPassthruFunction( 'esc_url' );
+
+		// Verify
+		$this->assertEquals( $expected, $manager->new_url( $original ) );
+	}
+
+	public function test_domain_replacement_mixed() {
+		$manager = DomainManager( 'http://test.com' );
+		$manager->add( 'https://cdn1.com', 'uploads' );
+
+		$original = 'http://test.com/image.png';
+		$expected = 'https://cdn1.com/image.png';
+
+		// Mocks
+		M::wpFunction( 'is_ssl', [ 'return' => false ] );
+		M::wpPassthruFunction( 'esc_url' );
+
+		// Verify
+		$this->assertEquals( $expected, $manager->new_url( $original ) );
+	}
+
+	public function test_has_domains() {
+		$manager = DomainManager( 'http://test.com' );
+		$this->assertFalse( $manager->has_domains( 'uploads' ) );
```

Comment:
wonder why this test wasn't working? isn't `uploads` the default value? or are you just adding this to be explicit.